x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/detail_panel/detail_panel.container.js

```diff
@@ -7,43 +7,23 @@
 import { connect } from 'react-redux';
 import { DetailPanel as DetailPanelView } from './detail_panel';
 
-import {
-  getDetailPanelAutoFollowPattern,
-  getDetailPanelAutoFollowPatternName,
-  getApiStatus,
-  isAutoFollowPatternDetailPanelOpen as isDetailPanelOpen,
-} from '../../../../../store/selectors';
-
-import {
-  closeAutoFollowPatternDetailPanel as closeDetailPanel,
-  editAutoFollowPattern,
-} from '../../../../../store/actions';
-
-import {
-  SECTIONS
-} from '../../../../../constants';
+import { getSelectedAutoFollowPattern, getSelectedAutoFollowPatternId, getApiStatus, } from '../../../../../store/selectors';
+import { selectAutoFollowPattern } from '../../../../../store/actions';
+import { SECTIONS } from '../../../../../constants';
 
 const scope = SECTIONS.AUTO_FOLLOW_PATTERN;
 
-const mapStateToProps = (state) => {
-  return {
-    isDetailPanelOpen: isDetailPanelOpen(state),
-    autoFollowPattern: getDetailPanelAutoFollowPattern(state),
-    autoFollowPatternName: getDetailPanelAutoFollowPatternName(state),
-    apiStatus: getApiStatus(scope)(state),
-  };
-};
-
-const mapDispatchToProps = (dispatch) => {
-  return {
-    closeDetailPanel: () => {
-      dispatch(closeDetailPanel());
-    },
-    editAutoFollowPattern: (name) => {
-      dispatch(editAutoFollowPattern(name));
-    }
-  };
-};
+const mapStateToProps = (state) => ({
+  autoFollowPatternId: getSelectedAutoFollowPatternId(state),
+  autoFollowPattern: getSelectedAutoFollowPattern(state),
+  apiStatus: getApiStatus(scope)(state),
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  // The actual closing of the panel is done in the <AutoFollowPatternList />
+  // component as it listens to the URL query change
+  closeDetailPanel: () => dispatch(selectAutoFollowPattern(null)),
```

Comment:
The logic might be a little easier to follow if the owner (`AutoFollowPatternList`) were to provide this callback instead of the container. Then the comment becomes a little less necessary because all of the related logic is already in front of you.